src/referencePicker/ReferencePickerModal.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+	<NcModal v-if="show"
+		:size="modalSize"
+		:can-close="false"
+		class="reference-picker-modal"
+		@close="onCancel">
+		<div class="reference-picker-modal--content">
+			<NcButton v-if="showBackButton"
+				:aria-label="backButtonTitle"
+				:title="backButtonTitle"
+				class="back-button"
+				@click="onBackClicked">
+				<template #icon>
+					<ArrowLeftIcon />
+				</template>
+			</NcButton>
+			<NcButton
+				:aria-label="closeButtonLabel"
+				:title="closeButtonTitle"
+				class="close-button"
+				@click="onCancel">
+				<template #icon>
+					<CloseIcon />
+				</template>
+			</NcButton>
+			<h2 v-if="showModalTitle">
+				{{ modalTitle }}
+			</h2>
+			<ReferencePicker
+				ref="referencePicker"
+				:initial-provider="initialProvider"
+				:focus-on-create="focusOnCreate"
+				@provider-selected="onProviderSelect"
+				@submit="onSubmit"
+				@cancel="onCancel" />
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import ArrowLeftIcon from 'vue-material-design-icons/ArrowLeft.vue'
+import CloseIcon from 'vue-material-design-icons/Close.vue'
+
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
```

Comment:
```suggestion
import { NcModal, NcButton } from '@nextcloud/vue'
```

Let's stick to the generic import in all places since this will make sure that `@nextcloud/vue` is not bundled within the vue-richtext package but apps can pull this in.

Reply:
NcHighlight needs to be directly imported, all the other ones are imported with the generic import now.